tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -14630,5 +14532,263 @@ SpirvEmitter::doUnaryExprOrTypeTraitExpr(const UnaryExprOrTypeTraitExpr *expr) {
   return sizeConst;
 }
 
+std::vector<SpirvInstruction *>
+SpirvEmitter::decomposeToScalars(SpirvInstruction *inst) {
+  QualType elementType;
+  uint32_t elementCount = 0;
+  uint32_t numOfRows = 0;
+  uint32_t numOfCols = 0;
+
+  QualType resultType = inst->getAstResultType();
+  if (hlsl::IsHLSLResourceType(resultType)) {
+    resultType = hlsl::GetHLSLResourceResultType(resultType);
+  }
+
+  // switch(inst->getAstResultType()->getKind())
+  if (isScalarType(resultType)) {
+    return {inst};
+  } else if (isVectorType(resultType, &elementType, &elementCount)) {
```

Comment:
Can you switch these else ifs after returns to ifs for readability? (https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html)

Same for the other examples below